### ISSUE TITLE
Masterbar: Unable to use some features on touch device

### ISSUE THREAD

**Issue by withinboredom** [Automattic/jetpack#8038]
#### Steps to reproduce the issue
Use a Windows computer (Chrome with touch features forced), or a mobile phone (Android/Chrome in this case)

To reproduce without a mobile device:
1. goto chrome://flags
2. set Touch Events API to Enabled from Automatic (this will send touch events instead of click events on pages that it would normally detect as non-touch friendly)
3. Then in Chrome dev tools, under sensors: set touch to be force enabled (mouse will turn into a circle instead of pointer when clicking on the page)

1. Visit wp-admin with wpcom Masterbar
2. The following features do not work:
  - Plan
 - Site Pages
 - Blog Posts
 - Comments
 - Themes
 - Sharing
 - People
 - Plugins
 - Settings
 - Switch Site

I should have just said that the only thing that works is Stats ...

#### What I expected
To navigate wp-admin from my mobile phone or touch screen

#### What happened instead
Unable to navigate

**Comment by jeherve** [comment]
Reopening as the navigation drawers were added back in #12454
